File: Syncfusion_Sample/Windows/Grid.Grouping.Windows/Samples/2.0/Filters and Expressions/Dynamic Filter Demo/vb/Localizer.vb
Imports Microsoft.VisualBasic
#Region "Copyright Syncfusion Inc. 2001 - 2013"
' Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
' Use of this code is subject to the terms of our license.
' A copy of the current license can be obtained at any time by e-mailing
' licensing@syncfusion.com. Any infringement will be prosecuted under
' applicable laws. 
#End Region
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Syncfusion.Windows.Forms
Imports Syncfusion.Windows.Forms.Grid
Imports Syncfusion.GridHelperClasses
Imports System.Threading

Namespace DynamicFilter
	Friend Class Localizer
		Implements ILocalizationProvider
		#Region "ILocalizationProvider Members"

		Private str As String = String.Empty
		Public Sub getstring(ByVal s As String)
			str = s
		End Sub
		Public Function GetLocalizedString(ByVal culture As System.Globalization.CultureInfo, ByVal name As String, ByVal obj As Object) As String Implements ILocalizationProvider.GetLocalizedString
			If str = "True" Then
				Select Case name
'					#Region "Menu Package"
					Case DynamicFilterResourceIdentifiers.StartsWith
						Return "empieza con"
					Case DynamicFilterResourceIdentifiers.EndsWith
						Return "termina con"
					Case DynamicFilterResourceIdentifiers.Equal
						Return "es igual a"
					Case DynamicFilterResourceIdentifiers.GreaterThan
						Return "mayor que"
					Case DynamicFilterResourceIdentifiers.GreaterThanOrEqualTo
						Return "Mayor o igual a"
					Case DynamicFilterResourceIdentifiers.LessThan
						Return "menos que"
					Case DynamicFilterResourceIdentifiers.LessThanOrEqualTo
						Return "Menor o igual a"
					Case DynamicFilterResourceIdentifiers.Like
						Return "como"
					Case DynamicFilterResourceIdentifiers.Match
						Return "partido"
					Case DynamicFilterResourceIdentifiers.NotEquals
						Return "no es igual"
					Case DynamicFilterResourceIdentifiers.ExpressionMATCH
						Return "expresión de coincidencia"
'					#End Region

					Case Else
						Return String.Empty
				End Select
			Else
				Select Case name
'					#Region "Menu Package"
					Case DynamicFilterResourceIdentifiers.StartsWith
						Return "StartsWith"
					Case DynamicFilterResourceIdentifiers.EndsWith
						Return "EndsWith"
					Case DynamicFilterResourceIdentifiers.Equal
						Return "Equals"
					Case DynamicFilterResourceIdentifiers.GreaterThan
						Return "GreaterThan"
					Case DynamicFilterResourceIdentifiers.GreaterThanOrEqualTo
						Return "GreaterThanOrEqualTo"
					Case DynamicFilterResourceIdentifiers.LessThan
						Return "LessThan"
					Case DynamicFilterResourceIdentifiers.LessThanOrEqualTo
						Return "LessThanOrEqualTo"
					Case DynamicFilterResourceIdentifiers.Like
						Return "Like"
					Case DynamicFilterResourceIdentifiers.Match
						Return "Match"
					Case DynamicFilterResourceIdentifiers.NotEquals
						Return "NotEquals"
					Case DynamicFilterResourceIdentifiers.ExpressionMATCH
                        Return "Expression MATCH"
'					#End Region

					Case Else
						Return String.Empty
				End Select
			End If
		End Function
		#End Region
	End Class
End Namespace

File: Syncfusion_Sample/Windows/Grid.Grouping.Windows/Samples/2.0/Filters and Expressions/Dynamic Filter Demo/cs/Localizer.cs
﻿#region Copyright Syncfusion Inc. 2001 - 2013
// Copyright Syncfusion Inc. 2001 - 2013. All rights reserved.
// Use of this code is subject to the terms of our license.
// A copy of the current license can be obtained at any time by e-mailing
// licensing@syncfusion.com. Any infringement will be prosecuted under
// applicable laws. 
#endregion
using System;
using System.Collections.Generic;
using System.Text;
using Syncfusion.Windows.Forms;
using Syncfusion.Windows.Forms.Grid;
using Syncfusion.GridHelperClasses;
using System.Threading;

namespace DynamicFilter
{
    class Localizer : ILocalizationProvider
    {
        #region ILocalizationProvider Members

        string str = string.Empty;
        public void getstring(string s)
        {
            str = s;
        }
        public string GetLocalizedString(System.Globalization.CultureInfo culture, string name,object obj)
        {
            if (str == "True")
            {
                switch (name)
                {
                    #region Menu Package
                    case DynamicFilterResourceIdentifiers.StartsWith:
                        return "empieza con";
                    case DynamicFilterResourceIdentifiers.EndsWith:
                        return "termina con";
                    case DynamicFilterResourceIdentifiers.Equal:
                        return "es igual a";
                    case DynamicFilterResourceIdentifiers.GreaterThan:
                        return "mayor que";
                    case DynamicFilterResourceIdentifiers.GreaterThanOrEqualTo:
                        return "Mayor o igual a";
                    case DynamicFilterResourceIdentifiers.LessThan:
                        return "menos que";
                    case DynamicFilterResourceIdentifiers.LessThanOrEqualTo:
                        return "Menor o igual a";
                    case DynamicFilterResourceIdentifiers.Like:
                        return "como";
                    case DynamicFilterResourceIdentifiers.Match:
                        return "partido";
                    case DynamicFilterResourceIdentifiers.NotEquals:
                        return "no es igual";
                    case DynamicFilterResourceIdentifiers.ExpressionMATCH:
                        return "expresión de coincidencia";
                    #endregion

                    default:
                        return string.Empty;
                }
            }
            else 
            {
                switch (name)
                {
                    #region Menu Package
                    case DynamicFilterResourceIdentifiers.StartsWith:
                        return "StartsWith";
                    case DynamicFilterResourceIdentifiers.EndsWith:
                        return "EndsWith";
                    case DynamicFilterResourceIdentifiers.Equal:
                        return "Equals";
                    case DynamicFilterResourceIdentifiers.GreaterThan:
                        return "GreaterThan";
                    case DynamicFilterResourceIdentifiers.GreaterThanOrEqualTo:
                        return "GreaterThanOrEqualTo";
                    case DynamicFilterResourceIdentifiers.LessThan:
                        return "LessThan";
                    case DynamicFilterResourceIdentifiers.LessThanOrEqualTo:
                        return "LessThanOrEqualTo";
                    case DynamicFilterResourceIdentifiers.Like:
                        return "Like";
                    case DynamicFilterResourceIdentifiers.Match:
                        return "Match";
                    case DynamicFilterResourceIdentifiers.NotEquals:
                        return "NotEquals";
                    case DynamicFilterResourceIdentifiers.ExpressionMATCH:
                        return "Expression MATCH";
                    #endregion

                    default:
                        return string.Empty;
                }
            }
        }
        #endregion
    }
}
